Jetcaster/app/src/main/java/com/example/jetcaster/ui/home/Home.kt

```diff
@@ -230,14 +231,73 @@ private fun getExcludedVerticalBounds(posture: Posture, hingePolicy: HingePolicy
     }
 }
 
-@OptIn(ExperimentalMaterial3AdaptiveApi::class)
 @Composable
 fun MainScreen(
     windowSizeClass: WindowSizeClass,
     navigateToPlayer: (EpisodeInfo) -> Unit,
     viewModel: HomeViewModel = hiltViewModel()
 ) {
-    val viewState by viewModel.state.collectAsStateWithLifecycle()
+    val homeScreenUiState by viewModel.state.collectAsStateWithLifecycle()
+    when (val uiState = homeScreenUiState) {
+        is HomeScreenUiState.Loading -> HomeScreenLoading()
+        is HomeScreenUiState.Error -> HomeScreenError(onRetry = viewModel::refresh)
+        is HomeScreenUiState.Ready -> {
+            HomeScreenReady(
+                uiState = uiState,
+                windowSizeClass = windowSizeClass,
+                navigateToPlayer = navigateToPlayer,
+                viewModel = viewModel,
+            )
+        }
+    }
+}
+
+@Composable
+private fun HomeScreenLoading(modifier: Modifier = Modifier) {
+    Surface(modifier.fillMaxSize()) {
+        Box {
+            CircularProgressIndicator(
```

Comment:
can you use instead some "loading components"? Like a grid/column of few grey items showing that content is loading?

Reply:
I think this relates to Tomas's comments below about intermediate loading/error states. I made a suggestion there on how to improve this so let's use that thread to discuss.